package.json

```diff
@@ -28,10 +28,14 @@
     "lint:worker": "gulp lint:worker",
     "format:worker": "gulp format:worker",
     "test": "npm run test:node && npm run test:worker",
-    "test:node": "gulp build && jest",
-    "test:worker": "gulp test:worker",
-    "coverage:node": "gulp build && jest --coverage && open-cli coverage/lcov-report/index.html",
-    "postinstall": "gulp build"
+    "test:node": "npm run postinstall && jest",
+    "test:worker": "run-script-os",
+    "test:worker:win32": "echo test:worker not support yet!!! run test in visual studio!!!",
+    "test:worker:default": "make test -C worker",
+    "coverage:node": "npm run postinstall && jest --coverage && open-cli coverage/lcov-report/index.html",
+    "postinstall": "run-script-os",
+    "postinstall:win32": "python ./worker/scripts/configure.py --format=msvs && IF %MEDIASOUP_BUILDTYPE% == Release ( MSBuild ./worker/mediasoup-worker.sln /p:Configuration=Release -t:mediasoup-worker ) ELSE ( MSBuild ./worker/mediasoup-worker.sln /p:Configuration=Debug -t:mediasoup-worker )",
```

Comment:
The problem here is that this line is trying to replace the whole `make` task in `worker/Makefile` which does much more that this. For instance, `make` allows using a custom `PYTHON` env variable to choose the preferred python executable. It also allows setting `MEDIASOUP_WORKER_BIN` env variable (for development purposes):

https://github.com/versatica/mediasoup/blob/v3/worker/Makefile#L20

May be we could create a separate shell/SuperShell/whatever script in `worker/` folder just for Windows and call it from `"potinstall:win32"? Something like:

```json
"postinstall:win32": "./worker/win-tasks.sh make"
```

(no idea about the script extension). Such a script (which is just for Windows) should take care of the PYTHON,  MEDIASOUP_BUILDTYPE  and MEDIASOUP_WORKER_BIN environment variables (if set) or use default values. It should also read the single argument it's called with (in this case "make"), do a `switch` internally and run the corresponding code. This is, something like a replacement for Windows of the `worker/Makefile` file.

Does this make sense?

Reply:
do not make much sense for me. already test both python2 and python3, those two work fine. **MEDIASOUP_BUILDTYPE** already checked in `postinstall:win32`. **MEDIASOUP_WORKER_BIN** is checked to skip some tasks if my understanding is right.

Reply:
If we want to replace all `make` tasks,  create a `./worker/win-make.js` seems a better solution. Shell script's syntax is complicated for me 😅.

Reply:
You are right, if `MEDIASOUP_WORKER_BIN` is set, Makefile tasks are skipped.

Regarding python 2 or 3, yes, it may work fine. The thing is that it can not be chosen by the user. In Linux/OSX the user can set `PYTHON=xxxx` env variable and then run `npm install` or `make` and `xxxx` will be used as python executable. That's not possible in Windows with the above solution.

> If we want to replace all `make` tasks,  create a `./worker/win-make.js` seems a better solution. Shell script's syntax is complicated for me 😅.

Oh, sure. I just don't know how it must be, but of course it would be cool to have this:

```json
"postinstall:win32": "$NODE ./worker/win-tasks.js make"
```

:)

Reply:
already had [a draft version ](https://github.com/haiyangwu/mediasoup-win/commit/7b2053498b1ea681f188d79613072c844428e51f), and the postinstall script is
`"postinstall:win32": "\"\"%NODE%\"\" ./worker/win-tasks.js make"`

but these two tasks `bear` and `tidy` used for what? I can't get the exact point. 😅

Reply:
Looks nice! I won't be able to properly review it until next week. For now just a few considerations:

* Forget about `bear`, `tidy` and other Makefile tasks. All them are [documented](https://github.com/versatica/mediasoup/blob/v3/doc/Building.md#makefile) but they must not be added to `win-tasks.js` (they are for development purposes).
* The style of the JS does not conform to mediasoup eslint rules. Try adding it to package.json:
```
"lint:node": "eslint -c .eslintrc.js gulpfile.js lib test worker/win-tasks.js",
```
* Let's not use "use strict". It's already implicit (it should produce an eslint warning in fact).

Reply:
@ibc updated.